src/main/java/com/notably/model/block/BlockTreeItemImpl.java

```diff
@@ -0,0 +1,146 @@
+package com.notably.model.block;
+
+import static java.util.Objects.hash;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.notably.model.block.exceptions.DuplicateBlockException;
+import com.notably.model.block.exceptions.NoSuchBlockException;
+
+import javafx.scene.control.TreeItem;
+
+/**
+ * Represents a BlockTreeItem in the Notably data structure.
+ */
+public class BlockTreeItemImpl implements BlockTreeItem {
```

Comment:
I think you don't really need fields for `parent` and `children`?

I believe the perhaps more correct way is to make the `getBlockParent` and `getBlockChildren` method call the underlying `blockTreeItem`'s `getParent` and `getChildren` methods. Of course, this will mean you'll need to convert the `TreeItem<Block>` instance to a `BlockTreeItem` instance, but I think that's not hard to achieve if you have a `public BlockTreeItemImpl(TreeItem<Block> treeItem)` constructor.

The reason for this argument basically boils down to two points:
1. We need to leverage on `TreeItem`'s observability behavior to make our `BlockTreeItem` observable too. The only way to do that is to make the wrapped `TreeItem` know about the tree structure, i.e. what's the parent and children of a `BlockTreeItem` (I believe you have already done this).
2. I'm afraid that duplicating `children` and `parent` (I mean duplicating because essentially these fields already exist in the wrapped `blockTreeItem`) might result in a bug-prone implementation. You'll need to always make sure that changes to one are reflected in the other, and vice versa.

Reply:
Actually, I think your implementation is working great, but perhaps it can be made simpler if you don't have to duplicate the `children` and `parent` states. Let me know what you think!

Reply:
Fine with me! I had the intention to keep `children` and `parent` separate for possible future change in implementation but I think that might be far-fetched (at least until core functionality is complete)